nano/node/bootstrap/bootstrap_service.cpp

```diff
@@ -566,14 +564,24 @@ void nano::bootstrap_service::run_one_priority ()
 	{
 		return;
 	}
-	auto [account, priority] = wait_priority ();
+	auto [account, priority, fails] = wait_priority ();
 	if (account.is_zero ())
 	{
 		return;
 	}
+
+	// Decide how many blocks to request
 	size_t const min_pull_count = 2;
-	auto count = std::clamp (static_cast<size_t> (priority), min_pull_count, nano::bootstrap_server::max_blocks);
-	request (account, count, channel, query_source::priority);
+	auto pull_count = std::clamp (static_cast<size_t> (priority), min_pull_count, nano::bootstrap_server::max_blocks);
+
+	bool sent = request (account, pull_count, channel, query_source::priority);
+
+	// Do not throttle accounts that are probably fully synced
+	if (sent && fails == 0)
+	{
+		nano::lock_guard<nano::mutex> lock{ mutex };
+		accounts.timestamp_set (account);
+	}
```

Comment:
By setting the timestamp to current value ( `accounts.timestamp_set (account);`  ) a cooldown period in `next_priority ()` is applied to this account. Is that the desired behavior  ?

Reply:
This should work as intended. It might be counterintuitive because we want to throttle only accounts with potentially more blocks; it's to avoid requesting blocks from the same frontier over and over before the current batch is processed by block processor.

The flow is something like:
```
if (likely to have more blocks)
{
set cooldown timestamp
}

// ... later in the block processing logic

if (last received block processed)
{
reset cooldown timestamp
}

Reply:
Yes, I updated the comment to make the intention clear